src/components/map-component.tsx

```diff
@@ -96,6 +101,13 @@ export class MapComponent extends BaseComponent<IProps, IState>{
     this.forceUpdate();
   }
 
+  public componentDidUpdate(prevProps: IProps) {
```

Comment:
My comments about `CrossSectionComponent` state management my apply to this component too, seems like a similar pattern.